Automattic/wp-calypso
# Changed use-url-search to use-query-args and added support to queryParams

## Changes proposed in this Pull Request

#### Added support to queryParams on `use-query-args` hook. 

Using object call (new feature): Pass one or more URL attributes: `{ s: 'seo', c: 'category name' }` , the `URL` will be http://calypso.localhost:3000/plugins?s=seo&c=category+name


#### Added support to get the current queryParams
Use the method `getQueryArgs` to get an object with the current URL attributes. For example, having the `URL` http://calypso.localhost:3000/plugins?s=seo&c=category+name we'll get this object as a result: `{ s: 'seo', c: 'category name' }`

## Testing instructions



* Open /plugins page and make sure everything related to search works as expected
* Customize the /plugins page adding some extra parameters, like `http://calypso.localhost:3000/plugins?s=seo&c=category+name` and make sure when clicking on the filter, only the `s` parameter will change its value.
* For testing the new getQueryArgs, I suggest adding the following code replacing the line 80 of the file  `client/my-sites/plugins/plugins-browser/index.jsx`: 
```
	const { setQueryArgs, getQueryArgs } = useQueryArgs();
	window.getQueryArgs = getQueryArgs;
```
Then, you can call it in the console and get the result:

![image](https://user-images.githubusercontent.com/1044309/169363675-7e697224-ef02-4791-8424-95de2d3a04e6.png)





Related to https://github.com/Automattic/wp-calypso/issues/63013
